File: src/main/java/miretz/ycloud/GuiceUIProvider.java
package miretz.ycloud;

import com.google.inject.Inject;
import com.google.inject.Injector;
import com.vaadin.server.UIClassSelectionEvent;
import com.vaadin.server.UICreateEvent;
import com.vaadin.server.UIProvider;

@SuppressWarnings("serial")
public class GuiceUIProvider extends UIProvider {

	@Inject
	private Injector injector;

	@Override
	public YCloudUI createInstance(UICreateEvent event) {
		return injector.getInstance(YCloudUI.class);
	}

	@Override
	public Class<YCloudUI> getUIClass(UIClassSelectionEvent event) {
		return YCloudUI.class;
	}
}

File: src/main/java/miretz/ycloud/GuiceUIProvider.kt
package miretz.ycloud

import com.google.inject.Inject
import com.google.inject.Injector
import com.vaadin.server.UIClassSelectionEvent
import com.vaadin.server.UICreateEvent
import com.vaadin.server.UIProvider

@SuppressWarnings("serial") class GuiceUIProvider : UIProvider() {

    @Inject
    private val injector: Injector? = null

    override fun createInstance(event: UICreateEvent): YCloudUI {
        return injector!!.getInstance(YCloudUI::class.java)
    }

    override fun getUIClass(event: UIClassSelectionEvent): Class<YCloudUI> {
        return YCloudUI::class.java
    }
}
